api/hasura/auth.ts

```diff
@@ -8,12 +8,24 @@ import { adminClient } from '../../api-lib/gql/adminClient';
 
 export default async function handler(req: VercelRequest, res: VercelResponse) {
   try {
-    if (IS_LOCAL_ENV && req.headers?.authorization === 'generate') {
-      // For generating libraries from inspection
-      res.status(200).json({
-        'X-Hasura-Role': req.headers?.['x-hasura-role'],
-      });
-      return;
+    if (IS_LOCAL_ENV) {
+      if (req.headers?.authorization === 'generate') {
+        // For generating libraries from inspection
+        res.status(200).json({
+          'X-Hasura-Role': req.headers?.['x-hasura-role'],
+        });
+        return;
+      }
+
+      // Skip auth checks when testing
+      if (req.headers?.authorization === 'test-skip-auth') {
```

Comment:
ideally I'd like this to be in a IS_TEST_ENV so this is really clearly only for when tests are running

Reply:
I've started implementing this suggestion with https://github.com/coordinape/coordinape/pull/1486/commits/86b2a76d0f7ab094d9979ae228ec898a58090da3, although looking at how nodemon is setup in `nodemon.json`:
```json
{
  "watch": [
    "api/",
    "api-lib/",
    "scripts/",
    "src/lib",
    "src/common-lib",
    "./*.json"
  ],
  "delay": 300,
  "execMap": {
    "ts": "ts-node --swc"
  },
  "events": { "start": "yarn typecheck:backend" },
  "env": { "NODE_ENV": "development" },
  "ignore": ["api-test/__tests__"]
}
```
it appears that it will always start in the `"development"` environment, and that's what gets executed when

`yarn exec nodemon \
  -- scripts/serve_dev.ts $PORT $PROXY_PORT`

Wondering if we need a new script for starting nodemon with a test environment just for tests, unsure how that would affect the app if at all. For now just added a placeholder with `export const IS_TEST_ENV = process.env.NODE_ENV === 'development' || 'test';`

Thoughts?

Reply:
I'm hoping once we figure out how the runner for this works, we can just pass in some ENV_VAR in the invocation of that runner that lets us know we're in a TEST env...